### DISCUSSION TITLE
Start GCODE

### DISCUSSION THREAD

**Discussion by jackpage43088** [mriscoc/Ender3V2S1#209] [Ideas]
If I used the start GCODE you provided does that mean the printer will determine the speed and flow rate and retraction? 

M201 X500.00 Y500.00 Z100.00 E5000.00 ;Setup machine max acceleration
M203 X500.00 Y500.00 Z20.00 E50.00 ;Setup machine max feedrate
M204 P500.00 R1000.00 T500.00 ;Setup Print/Retract/Travel acceleration
M205 X8.00 Y8.00 Z0.40 E5.00 ;Setup Jerk
M220 S100 ;Reset Feedrate
M221 S100 ;Reset Flowrate

I’m kinda new but does it mean it’s going off the eeprom settings saved in the Pinter firmware? does that mean it doesn’t matter what I put in the slicer as far as flow settings and retraction and speed goes?

**Comment by mriscoc** [comment]
The configuration in the Slicer has priority, but the start G-code sets some limits.

**Reply by csanii123** [reply]
There are 3 different settings (start gcode, slicer gcode and printer settings on the machine itself set with the dial) which will the printer actually use for the print ?
ie. The default max feedrate for Z in the firmware is 10, the start gcode in jackpage43088 example has it set to 20 and if I add a gcode line in my slicer to make it 15 which will the printer use for the print or is it always the last code the printer gets ?

Printer setting in EEPROM is always read 1st into memory. Then if there is start gcode it would "overwrite" what is in memory but not EEPROM for the print and then if there is a slicer setting after that it would "overwrite" what is in memory but not in EEPROM and that would take priority over the other 2 since it was the last setting made.  Is this correct ?
